storage/pipeline/types.go

```diff
@@ -158,7 +158,7 @@ func (t *SectorInfo) existingPieceSizes() []abi.UnpaddedPieceSize {
 
 func (t *SectorInfo) hasDeals() bool {
 	for _, piece := range t.Pieces {
-		if piece.HasDealInfo() {
+		if piece.HasDealInfo() && piece.HasDealProposal() {
```

Comment:
Why? HasDeals is meant to return true whenever the sector has any non-padding / non-cc piece. DDO pieces in this context are deal pieces.